### DISCUSSION TITLE
Problem on validating machine file using keygen-go

### DISCUSSION THREAD

**Discussion by 14mdzk** [keygen-sh/community#160] [Issue]
Hi, currently I working on go application that will be distributed offline thus after some browse into the docs, using **Machine File** is the best choices since it enables you to get both fingerprints and license at the same time. 

I have already following the docs provided in the [offline license files](https://github.com/keygen-sh/keygen-go?tab=readme-ov-file#offline-license-files) and it works as intended, but when I tried to change it into **Machine File** it keeps gave an error said that ``machine file is invalid``

This happened either I checked out the file through the [code](https://github.com/keygen-sh/keygen-go?tab=readme-ov-file#license-activation) nor from the dashboard app, is there something that I missed?

Many Thanks!

**Comment by ezekg** [comment]
If a license file works but a machine file doesn't, when verifying the machine file, make sure you change the `license/` signing prefix 
 to `machine/`.

**Reply by 14mdzk** [reply]
Yes, afaik the library already do so when verifying the machine file, [here](https://github.com/keygen-sh/keygen-go/blob/master/verifier.go) it is.

**Reply by ezekg** [reply]
Are you copy/pasting the license file, or using the license file downloaded directly from Keygen? The Go SDK expects a direct download from Keygen. Sometimes, when you copy/paste, newlines are changed from `\n` to `\r\n` e.g. on Windows. It looks like the Go SDK doesn't handle Windows line endings yet (i.e. carriage return and newline).

Opened an issue: https://github.com/keygen-sh/keygen-go/issues/24

**Reply by 14mdzk** [reply]
Yes, I tried both though it still recognized as invalid license file. That's most likely the case here.
Seems like I will use online validation approach as of now, thanks for the help!

**Comment by 14mdzk** [comment]
Apart from the problem that I described above, do you have any idea as to why when trying to validating the license it returned response `response signature is invalid`? This occurred when I trying to implement the CE in the cloud environment.

Thanks!

**Reply by ezekg** [reply]
@14mdzk are you using the Go SDK, or a custom intregration? You need to provide more information. If you're using the Go SDK, then your public key may be wrong. You can find your Keygen Cloud public key [here](https://app.keygen.sh/settings), or you can get your Keygen CE public key via running `Account.sole.ed25519_public_key` [in a console](https://keygen.sh/docs/self-hosting/#opening-a-console). If you're using a custom integration, there could be any number of things wrong with your signature verification logic. You will need to either be more specific or provide some code that I can review.

**Reply by 14mdzk** [reply]
Hi @ezekg, 

My apologize for so late reply, I have followed all the instructions and it went well for the local setup and I already found the culprit.
This happened because the machine I'm using isn't configured the **time setting** correctly which is why the signature won't match.

Thanks for the support!
